### PR TITLE
Upgrading Kafka Extension from 3.4.0 to 3.5.0

### PR DESCRIPTION
**Pull request by shrohilla**
### Issue describing the changes in this PR

resolves :-

1. https://github.com/Azure/azure-functions-kafka-extension/issues/326
2. https://github.com/Azure/azure-functions-dotnet-worker/issues/848
3. https://github.com/Azure/azure-functions-kafka-extension/issues/272
4. https://github.com/Azure/azure-functions-kafka-extension/issues/263
6. https://github.com/Azure/azure-functions-kafka-extension/issues/122
7. https://github.com/Azure/azure-functions-kafka-extension/issues/249

As feature wise :-

1. Added the improvement for the producer logging to app insights & filesystem 
2. Added the support for the retry feature
3. Added the support for keys for triggers & output bindings
4. Added the support for auto.offset.reset config 

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Comment by fabiocav** [comment]
@shrohilla Changes look good, but would it be possible to update the PR and commit descriptions to clarify this is an update to Kafka? Thanks!

**Comment by shrohilla** [comment]
> @shrohilla Changes look good, but would it be possible to update the PR and commit descriptions to clarify this is an update to Kafka? Thanks!

@fabiocav Updated in the description section

**Comment by kshyju** [comment]
@shrohilla Can you make a copy of release_notes.md from https://github.com/Azure/azure-functions-dotnet-worker/tree/main/sdk location and paste it in the [root of the kafka extension](**url**) and update the release notes as needed(Include information about this 3.5.0 release).

**Comment by kshyju** [comment]
> > @shrohilla Changes look good, but would it be possible to update the PR and commit descriptions to clarify this is an update to Kafka? Thanks!
> 
> @fabiocav Updated in the description section

Could you please update the PR title as well to include the extension information?

**Review by kshyju — APPROVED** [review]
Please update the PR title to be more elaborate.

**Comment by shrohilla** [comment]
> Please update the PR title to be more elaborate.

Done

**Comment by shrohilla** [comment]
> > > @shrohilla Changes look good, but would it be possible to update the PR and commit descriptions to clarify this is an update to Kafka? Thanks!
> > 
> > 
> > @fabiocav Updated in the description section
> 
> Could you please update the PR title as well to include the extension information?

Done

**Comment by shrohilla** [comment]
> @shrohilla Can you make a copy of release_notes.md from https://github.com/Azure/azure-functions-dotnet-worker/tree/main/sdk location and paste it in the [root of the kafka extension](**url**) and update the release notes as needed(Include information about this 3.5.0 release).

Added that please share your comments in-case it's not upto the mark

File: extensions/Worker.Extensions.Kafka/release_notes.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Release notes
+<!-- Please add your release notes in the following format:
+- My change description (#PR/#issue)
+-->
+
+### 3.5.0
+- Added the improvement for the producer logging to app insights & filesystem.
+- Added the support for the retry feature
+- Added the support for keys for triggers & output bindings
+- Added the support for auto.offset.reset config
```
Comment (kshyju) [inline extensions/Worker.Extensions.Kafka/release_notes.md:10]: Thanks! Feel free to add a link to the issue/PR (see line 3 for format).

Comment (shrohilla) [reply extensions/Worker.Extensions.Kafka/release_notes.md:10]: Done

File: extensions/Worker.Extensions.Kafka/release_notes.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+## Release notes
+<!-- Please add your release notes in the following format:
+- My change description (#PR/#issue)
+-->
+
+### 3.5.0
+- Added the improvement for the producer logging to app insights & filesystem. (https://github.com/Azure/azure-functions-kafka-extension/pull/377)
```
Comment (kshyju) [inline extensions/Worker.Extensions.Kafka/release_notes.md:7]: @shrohilla Could you change the link to just the PR id prefixed with #. Ex (#377)
See the SDK release notes for reference: https://github.com/Azure/azure-functions-dotnet-worker/edit/main/sdk/release_notes.md

Comment (shrohilla) [reply extensions/Worker.Extensions.Kafka/release_notes.md:7]: not sure what exactly are you asking but it's not possible for cross repo and as per my understanding it looks fine to me.

image

**Pull request merged**